Components/Metrics/PointToSurfaceDistance/elxPointToSurfaceDistanceMetric.h

```diff
@@ -0,0 +1,172 @@
+/*=========================================================================
+ *
+ *  Copyright UMC Utrecht and contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef __elxPointToSurfaceDistanceMetric_H__
+#define __elxPointToSurfaceDistanceMetric_H__
+
+#include "elxIncludes.h" // include first to avoid MSVS warning
+#include "itkPointToSurfaceDistanceMetric.h"
+
+namespace elastix
+{
+
+/**
+ *
+ * For more information check the paper:\n
+ * Gunay, G. , Luu, M. H., Moelker, A. , Walsum, T. and Klein, S. (2017), 
+ * "Semiautomated registration of pre‐ and intraoperative CT for image‐guided percutaneous 
+ * liver tumor ablation interventions," Med. Phys., 44: 3718-3725.
+ *
+ * The parameters used in this class are:\n
+ * \parameter Metric: Select this metric as follows:
+ *    </tt>(Metric "PointToSurfaceDistance")</tt>
+ *
+ * \parameter PointToSurfaceDistanceAverage: A parameter to get average of the points by dividing the weight of the penalty with the number of the points included. 
+ *    example: </tt>(PointToSurfaceDistanceAverage "true")</tt> \n
+ *    Default is "true".
+ *
+ * \Command line parameters
+ * \parameter -fp    "file" : Elastix pointset file. 
+ *  example: 
+
+ * \parameter -dt    "file" : Distance transform (file) of segmentation of the organ under interest in the moving image.
+ * \parameter -seg   "file" : Segmentation (file) of the organ under interest in the moving image. With this parameter the algorithm internally computes the distance transform.
+ * \parameter -dtout "file" : If the user wants to get the distance transform from organ segmentation, this parameter is called with the name of the distance transform image file to write .
+ *
+ *
+ * \ingroup Metrics
+ *
+ */
+
+template< class TElastix >
+class PointToSurfaceDistanceMetric :
+  public
+  itk::PointToSurfaceDistanceMetric <
+  typename MetricBase< TElastix >::FixedPointSetType,
+  typename MetricBase< TElastix >::MovingPointSetType >,
+  public MetricBase< TElastix >
+{
+public:
+
+  using Self = PointToSurfaceDistanceMetric;
+  /** The private constructor. */
+  PointToSurfaceDistanceMetric(const Self &) = delete;
+  /** The private copy constructor. */
+  void operator=(const Self &) = delete;
```

Comment:
Please note: `PointToSurfaceDistanceMetric(const Self &)` is not a _private_ constructor, and `operator=(const Self &)` is an assignment operator, _not_ a copy constructor. Personally I would do `ITK_DISALLOW_COPY_AND_ASSIGN(PointToSurfaceDistanceMetric)` instead.

Reply:
The comments just remaining from previous code that I forgot to modify. Can we use ITK_DISALLOW_COPY_AND_ASSIGN in elastix code?

Reply:
@gohkangg I realize I can't blame you for the incorrect comment, because it is there already in previous elastix code! `ITK_DISALLOW_COPY_AND_ASSIGN` is part of the public interface of ITK, and elastix uses ITK already, so yes, we can use it  😃

Reply:
Done ;)